pkg/geoipupdate/database/http_reader.go

```diff
@@ -59,58 +60,122 @@ func (reader *HTTPDatabaseReader) Get(destination Writer, editionID string) erro
 		url.QueryEscape(destination.GetHash()),
 	)
 
-	req, err := http.NewRequest(http.MethodGet, maxMindURL, nil) // nolint: noctx
-	if err != nil {
-		return errors.Wrap(err, "error creating request")
-	}
-	req.SetBasicAuth(fmt.Sprintf("%d", reader.accountID), reader.licenseKey)
-
 	if reader.verbose {
 		log.Printf("Performing update request to %s", maxMindURL)
 	}
-	response, err := internal.MaybeRetryRequest(reader.client, reader.retryFor, req)
+
+	var modified bool
+	// It'd be nice to not use a temporary file here. However the Writer API does
+	// not currently support multiple attempts to download the file (it assumes
+	// we'll begin writing once). Using a temporary file here should be a less
+	// disruptive alternative to changing the API. If we change that API in the
+	// future, adding something like Reset() may be desirable.
+	tempFile, err := ioutil.TempFile("", "geoipupdate")
 	if err != nil {
-		return errors.Wrap(err, "error performing HTTP request")
+		return errors.Wrap(err, "error opening temporary file")
 	}
 	defer func() {
-		if err := response.Body.Close(); err != nil {
-			log.Fatalf("Error closing response body: %+v", errors.Wrap(err, "closing body"))
+		if err := tempFile.Close(); err != nil {
+			log.Printf("error closing temporary file: %s", err)
 		}
-	}()
-
-	if response.StatusCode == http.StatusNotModified {
-		if reader.verbose {
-			log.Printf("No new updates available for %s", editionID)
+		if err := os.Remove(tempFile.Name()); err != nil {
+			log.Printf("error removing temporary file: %s", err)
 		}
-		return nil
+	}()
+	var newMD5 string
+	var modificationTime time.Time
+	err = internal.RetryWithBackoff(
+		func() error {
```

Comment:
Given the size, it might be a bit cleaner to move this to a private named function rather than using an anonymous function. I suppose you are doing it so that you can modify the local variables in `Get`, but it is pretty hard to follow. Perhaps a named function called by an anonymous function so that it is clear what local variables are getting updated?

Reply:
Yeah, that'll likely be nicer!